internal/operands/common-templates/reconcile.go

```diff
@@ -113,6 +120,30 @@ func incrementTemplatesRestoredMetric(reconcileResults []common.ReconcileResult,
 	}
 }
 
+func (c *commonTemplates) manageSingleReplicaInfrastructure(request *common.Request) {
+	if request.TopologyMode != v1.SingleReplicaTopologyMode {
+		return
+	}
+
+	for templateIdx, template := range c.templatesBundle {
+		for objIdx, rawObj := range template.Objects {
+			vm := &v12.VirtualMachine{}
+			err := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(rawObj.Raw), 1024).Decode(vm)
+			if err != nil {
+				continue
```

Comment:
Do we want to continue if there is some error?

Reply:
Since `Objects` are `runtime.RawExtension` I think that it is not guaranteed that they are all `VirtualMachine`. It should be, but in case they are not we can skip it IMO. If you think that it is better to block it I can first Decode to `Unstructured`, check if it is a VM, then Decode to VM and check if an error occurs. Or maybe the assumption that Objects can contain something different from VM is wrong and so I can simply raise an error here. WDYT? Thanks

Reply:
If the object is not a VM, then I think it is ok to skip it.
But it has to be decoded into `Unstructured` in any case, because the VM object in template may not be correct according to its schema. Templates can have variables that are substituted. The variable names are strings and they may be in places where `VirtualMachine` struct contains a different type, so the decode would fail.

Reply:
Nice point! You're right. I'm going to adjust the decode and work with `Unstructured`, thanks